cranelift/filetests/src/test_wasm.rs

```diff
@@ -0,0 +1,110 @@
+//! Test runner for `.wat` files to exercise CLIF-to-Wasm translations.
+
+mod config;
+mod env;
+
+use anyhow::{bail, ensure, Context, Result};
+use config::TestConfig;
+use env::ModuleEnv;
+use similar::TextDiff;
+use std::{fmt::Write, path::Path};
+
+/// Run one `.wat` test.
+pub fn run(path: &Path, wat: &str) -> Result<()> {
+    debug_assert_eq!(path.extension().unwrap_or_default(), "wat");
+
+    // The test config source is the leading lines of the WAT file that are
+    // prefixed with `;;!`.
+    let config_lines: Vec<_> = wat
+        .lines()
+        .take_while(|l| l.starts_with(";;!"))
+        .map(|l| &l[3..])
+        .collect();
+    let config_text = config_lines.join("\n");
```

Comment:
Is it worth trying to handle empty lines before the config section, or raising an error if config comments are found elsewhere in the test?

Reply:
I was trying to keep this code as dead simple as possible, and I think it is worth riding that as far as we can. If it starts becoming annoying, then we can start beefing this up.